Make community_list endpoint support required functionality

anttimaki opened the pull request:
Moving from placeholder implementation to the real deal on the client side has revealed some shortcomings in the functionality and the returned data of this endpoint.

- Add support for varying page size via page_size query param
- Add support for filtering based on name and description fields via search query param
- Add support for ordering by name and datetime_created fields
- Add datetime_created and icon_url to serializer
- Fields that can return null due to having Null=True in model definition are marked as optional in the serializer
- Add naive implementations for Community's total_package_count and total_downloads_count properties
- Add less naive implementation for icon_url property

Refs TS-1815

MythicManiac commented on django/thunderstore/api/cyberstorm/views/community_list.py:
```diff
@@ -7,14 +8,17 @@
 
 
 class CommunityPaginator(PageNumberPagination):
+    max_page_size = 100
     page_size = 100
+    page_size_query_param = "page_size"
```
Would strongly prefer fixed page sizes to reduce cache complexity, and I don't think we have a need to use non-standard one?

anttimaki replied:
Yeah I think we can live without it for now. I just added the support here since it was present on the frontend before I started working on it. Only place where it would be needed in addition to the community list would have been the landing page where a small batch top communities are shown, but I'm not sure if that would be handled by this endpoint anyway. (And the frontend implementation didn't even use pagination, but sliced the result set on client 🙄)

MythicManiac commented on django/thunderstore/community/models/community.py:
```diff
@@ -66,12 +66,24 @@ class Community(TimestampMixin, models.Model):
         # TODO: Implement as a cached value with background updates
-        return -1
+        listings = self.package_listings.active()
+
+        if self.require_package_listing_approval:
+            listings = listings.approved()
+
+        return listings.count()
 
     @property
     def total_download_count(self) -> int:
         # TODO: Implement as a cached value with background updates
-        return -1
+        # TODO: also figure out more efficient way to handle this
+
+        listings = self.package_listings.active()
+
+        if self.require_package_listing_approval:
+            listings = listings.approved()
+
+        return sum([l.total_downloads for l in listings])
```
Can we add a check here which returns `-1` on non-debug environments until the cache-based solution exists? Otherwise it'll probably end up blocking deployment as it's a potential DOS vector I wouldn't feel particularly good about deploying.

anttimaki commented:
- removed `page_size_query_param` and `max_page_size`
- changed `page_size` to 150 to circumvent the lack of pagination on client side, for now
- added the debug check as requested and updated/augmented unit tests
